plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetescluster/KubernetesClusterManagerImpl.java

```diff
@@ -0,0 +1,3061 @@
+                                        "tcp", networkId, accountId, domainId, vmIdFinal);
+                        newRule.setDisplay(true);
+                        newRule.setState(FirewallRule.State.Add);
+                        newRule = portForwardingRulesDao.persist(newRule);
+                        return newRule;
+                    }
+                });
+                rulesService.applyPortForwardingRules(publicIp.getId(), account);
+                LOGGER.debug(String.format("Provisioned SSH port forwarding rule from port %d to 22 on %s to the VM IP: %s in Kubernetes cluster ID: %s", srcPortFinal, publicIp.getAddress().addr(), vmIp, kubernetesCluster.getUuid()));
+            } catch (Exception e) {
+                String msg = String.format("Failed to activate SSH port forwarding rules for the Kubernetes cluster ID: %s", kubernetesCluster.getUuid());
+                LOGGER.warn(msg, e);
+                stateTransitTo(kubernetesCluster.getId(), KubernetesCluster.Event.CreateFailed);
+                throw new ManagementServerException(msg, e);
+            }
+        }
+    }
+
+    // Open up  firewall ports NODES_DEFAULT_START_SSH_PORT to NODES_DEFAULT_START_SSH_PORT+n for SSH access. Also create port-forwarding
+    // rule to forward public IP traffic to all node VM private IP. Existing node VMs before scaling
+    // will already be having these rules
+    private void scaleKubernetesClusterNetworkRules(KubernetesCluster kubernetesCluster, Network network, Account account,
+                                                    List<Long> clusterVMIds, List<Long> removedVMIds) throws ManagementServerException {
+        if (!Network.GuestType.Isolated.equals(network.getGuestType())) {
+            LOGGER.debug(String.format("Network ID: %s for Kubernetes cluster ID: %s is not an isolated network, therefore, no need for network rules", network.getUuid(), kubernetesCluster.getUuid()));
+            return;
+        }
+        IpAddress publicIp = null;
+        List<? extends IpAddress> addresses = networkModel.listPublicIpsAssignedToGuestNtwk(network.getId(), true);
+        if (CollectionUtils.isEmpty(addresses)) {
+            String msg = String.format("No public IP addresses found for network ID: %s, Kubernetes cluster ID: %s", network.getUuid(), kubernetesCluster.getUuid());
+            LOGGER.error(msg);
+            throw new ManagementServerException(msg);
+        }
+        for (IpAddress address : addresses) {
+            if (address.isSourceNat()) {
+                publicIp = address;
+                break;
+            }
+        }
+        if (publicIp == null) {
+            String msg = String.format("No source NAT IP addresses found for network ID: %s, Kubernetes cluster ID: %s", network.getUuid(), kubernetesCluster.getUuid());
+            LOGGER.error(msg);
+            throw new ManagementServerException(msg);
+        }
+
+        List<String> sourceCidrList 
```

Comment:
createKubernetesCluster(CreateKubernetesClusterCmd cmd) should do nothing more than instantiate  a worker class that has access to all  the utilities it needs to create a K8s cluster and let it do it's work.